### PR TITLE
vscode-extensions: add ms-python.flake8 and ms-python.pylint, update ms-python.black-formatter

### PR DESCRIPTION
**Pull request by amadejkastelic**
## Description of changes

https://marketplace.visualstudio.com/items?itemName=ms-python.flake8
https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter
https://marketplace.visualstudio.com/items?itemName=ms-python.pylint

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -3190,6 +3190,21 @@ let
...
       };
 
+      ms-python.flake8 = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "flake8";
+          publisher = "ms-python";
+          version = "2023.10.0";
+          hash = "sha256-4Vjw8yJPrxLg0hcoTw8AEBEcmQ9sEUNqFaHLxICks0E=";
+        };
+        meta = with lib; {
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3200]:
```suggestion
        meta = {
```
- Please, don't introduce nested with.
  https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/editors/vscode/extensions#conventions-for-adding-new-extensions

- Changelog is missing.

- Consider adding yourself as maintainer.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -3178,15 +3178,19 @@ let
...
           description = "Formatter extension for Visual Studio Code using black";
           downloadPage = "https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter";
           homepage = "https://github.com/microsoft/vscode-black-formatter";
-          license = licenses.mit;
-          maintainers = with maintainers; [ sikmir ];
+          license = lib.licenses.mit;
+          maintainers = [
+            lib.maintainers.amadejkastelic
+            lib.maintainers.sikmir
+          ];
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3193]: > maintainers = with lib.maintainers; [ ]

FYI, this pattern is acceptable. But what you have done also is. I'm not asking for changing anything, just informing of the possibilities.

**Comment by superherointj** [comment]
Please change the maintainer commit title to:

> maintainers: add amadejkastelic

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -3178,15 +3178,36 @@ let
...
```
Comment (drupol) [inline pkgs/applications/editors/vscode/extensions/default.nix:3193]:
```suggestion
          maintainers = with lib.maintainers; [
            amadejkastelic
            sikmir
          ];
```

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 328632` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>vscode-extensions.ms-python.black-formatter</li>
    <li>vscode-extensions.ms-python.flake8</li>
    <li>vscode-extensions.ms-python.pylint</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**